client/allocrunner/taskrunner/envoybootstrap_hook.go

```diff
@@ -76,19 +78,25 @@ func (h *envoyBootstrapHook) Prestart(ctx context.Context, req *interfaces.TaskP
 	//     the host netns.
 	grpcAddr := "unix://" + allocdir.AllocGRPCSocket
 
+	// Envoy runs an administrative API on the loopback interface. If multiple sidecars
+	// are running, the bind addresses need to have unique ports.
+	// TODO: support running in host netns, using freeport to find available port
```

Comment:
I wonder if there's something clever we can do to let the OS pick the port, and then lookup what port got picked after the sidecar process launch. Even the updated freeport is a fingers-crossed kind of thing

Reply:
Agreed, let's remove this TODO as I don't think freeport would be a significant improvement. Since this is only on localhost within a netns the only potential conflict is with other services using localhost in the netns (likely the task the sidecar is proxying for). We know the port we're proxying to and can validate it won't conflict.

That leaves the only remaining potential conflict as internal-only ports on other tasks in the netns which should be rare. I think if we document this it will hopefully prevent errors.

Does Envoy exit and log with a reasonable error message when this condition is hit so at least people stand a chance of diagnosing it for themselves?

Reply:
yes envoy will log a bind error to stderr.

However this todo was specifically for when we support connect without network namespaces (host ns). We won't be able to use the current alloc local incrementing port as all allocs in host networking mode would be sharing the loopback.